### ISSUE TITLE
decoding \'{e} and \'\i work but \'{\i} doesn't

### ISSUE THREAD

**Issue by michamos** [mcmtroffaes/latexcodec#58]
Hi,

it looks like there is a bug when decoding diacritics followed by special characters:

```
In [1]: import latexcodec

In [2]: print(u"\\'{e}-\\'\\i-\\'{\\i}".decode('latex'))
{é}-í-\'{ı}
```
I would expect the last one to give `{í}` instead. 

I tried to fix it by relaxing https://github.com/mcmtroffaes/latexcodec/blob/b1ee5dbe71f295e03500681a0e7baf49aa47ff01/latexcodec/codec.py#L616 to also include `control_word`, but it doesn't seem to be so simple.

**Comment by michamos** [comment]
@mcmtroffaes it's been some time now. Do you have any input?

**Comment by mcmtroffaes** [comment]
Sorry for the long silence. This is indeed not so simple. I've looked into the details, and to see if this could be implemented. The encoder works by matching the last few tokens in the token buffer, doing a translation (and flushing the buffer) as soon as it finds a match (with a preference for longer token combinations). So, at the moment "\\'{\\i" is in the buffer, the "\\i" will match and will be translated, even if tokens for "\\'{\\i}" were to be added to the translation map... In essence, the codec does not do full parsing, and therefore cannot match the pattern in the current design.

So, in a nutshell, if you want ``u"í"``, you must use ``b"\\'\\i"``...
